rollingpin/main.py

```diff
@@ -153,9 +162,18 @@ def _main(reactor, *raw_args):
     # the cypher used for SSH connections. we don't care and can't do anything
     # about it for now.
     warnings.simplefilter("ignore")
+    initial_parser = make_profile_parser(PROFILE_DIRECTORY)
+    if not raw_args:
+        initial_parser.print_help()
+        sys.exit(0)
+    args, raw_args = initial_parser.parse_known_args(args=raw_args)
+
+    config = _load_configuration(args.profile, PROFILE_DIRECTORY)
+    args = _parse_args(config, raw_args, args.profile)
+    if args.test:
```

Comment:
Followed up on this, current output looks like:
```
root@rollingpin.lxc:# rollout example --test -c bounce
rollout -h apps --parallel=2 --timeout=60 -d rollingpin -r all -c bounce

This will:

* Deploy the `rollingpin` component.
* Restart `all` applications.
* Run the `bounce` command.

on `apps` hosts, 2 at a time, timing out if a host takes more than 60 seconds
```

This definitely is not taking into account some of the less common flags (such as `--startat`, `--stopbefore`, `--no-harold`, etc) but should give similar output to what Neil suggested.